From 60d9623abade20f7749d8bfc0f694a19783a092e Mon Sep 17 00:00:00 2001
Message-Id: <60d9623abade20f7749d8bfc0f694a19783a092e.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Thu, 12 Feb 2015 11:13:17 +0530
Subject: [PATCH 58/61] FOR_UPSTREAM [VPG]: drm/i915/chv: Correcting the GPIO
 mapping.

The CHV GPIO pin numbers were mapped incorrectly for the
South-east and east core. Hence correcting the same in
this patch.

Issue: GMINL-5197
Change-Id: I648dcf2989fe5a60fbf682f5ba681929e558f88a
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h            |    6 +++---
 drivers/gpu/drm/i915/intel_dsi_panel_vbt.c |   10 +++++-----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index e6ea3bf..98ca343 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -532,11 +532,11 @@
 #define   MAX_GPIO_NUM_SC			128
 #define   MAX_GPIO_NUM				172
 #define   CHV_MAX_GPIO_NUM_N			72
-#define   CHV_MAX_GPIO_NUM_SE			99
+#define   CHV_MAX_GPIO_NUM_E			99
 #define   CHV_MAX_GPIO_NUM_SW			197
-#define   CHV_MIN_GPIO_NUM_SE			73
+#define   CHV_MIN_GPIO_NUM_E			73
 #define   CHV_MIN_GPIO_NUM_SW			100
-#define   CHV_MIN_GPIO_NUM_E			198
+#define   CHV_MIN_GPIO_NUM_SE			198
 #define   IOSF_PORT_CCK				0x14
 #define   IOSF_PORT_CCU				0xA9
 #define   IOSF_PORT_GPS_CORE			0x48
diff --git a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
index afe19bc..9b44df1 100644
--- a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
+++ b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
@@ -334,17 +334,17 @@ static int chv_program_gpio(struct intel_dsi *intel_dsi,
 		if (gpio <= CHV_MAX_GPIO_NUM_N) {
 			block = CHV_IOSF_PORT_GPIO_N;
 			DRM_DEBUG_DRIVER("GPIO is in the north Block\n");
-		} else if (gpio <= CHV_MAX_GPIO_NUM_SE) {
-			block = CHV_IOSF_PORT_GPIO_SE;
-			gpio = gpio - CHV_MIN_GPIO_NUM_SE;
+		} else if (gpio <= CHV_MAX_GPIO_NUM_E) {
+			block = CHV_IOSF_PORT_GPIO_E;
+			gpio = gpio - CHV_MIN_GPIO_NUM_E;
 			DRM_DEBUG_DRIVER("GPIO is in the south east Block\n");
 		} else if (gpio <= CHV_MAX_GPIO_NUM_SW) {
 			block = CHV_IOSF_PORT_GPIO_SW;
 			gpio = gpio - CHV_MIN_GPIO_NUM_SW;
 			DRM_DEBUG_DRIVER("GPIO is in the south west Block\n");
 		} else {
-			block = CHV_IOSF_PORT_GPIO_E;
-			gpio = gpio - CHV_MIN_GPIO_NUM_E;
+			block = CHV_IOSF_PORT_GPIO_SE;
+			gpio = gpio - CHV_MIN_GPIO_NUM_SE;
 			DRM_DEBUG_DRIVER("GPIO is in the east Block\n");
 		}
 	} else
-- 
1.7.9.5

